### ISSUE TITLE
Disable editing more fields of `controlplane` group

### ISSUE THREAD

**Issue by AlanCoding** [ansible/awx#11130]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](http://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I am not entitled to status updates or other assurances.

### Summary

Users are able to modify several fields of the `controlplane` instance group, which they probably shouldn't be allowed to edit (these edits will break things).

### AWX version

19.2.2

### Installation method

docker on linux

### Modifications

no

### Ansible version

devel

### Operating system

N/A

### Web browser

Chrome

### Steps to reproduce

PATCH to `/api/v2/instance_groups/N/` where that record has the `name` of "controlplane"

try giving all the fields (individually) a non-default value:

```json
    "name": "controlplane",
    "is_container_group": false,
    "credential": null,
    "policy_instance_percentage": 98,
    "policy_instance_minimum": 0,
    "policy_instance_list": [],
    "pod_spec_override": ""
```

### Expected results

Most of these should give a 400 error

### Actual results

Only modifying `name` gives a 400

Out of the others, particularly problematic seeming ones are:

 - I can set `is_container_group` to true... that doesn't sound good
 - I can set `policy_instance_percentage` to a value lower than what was set by the installer, which is probably not desired (maybe arguable? but I doubt it)
 - I can add execution nodes to `policy_instance_list`, which is absolutely not okay

**Comment by AlanCoding** [comment]
existing name validation

https://github.com/ansible/awx/blob/6a17e5b65bf9421c2e650d23817c79ce81248c6f/awx/api/serializers.py#L4935

**Comment by AlanCoding** [comment]
It would be really easy to deal with this along with:

https://github.com/ansible/awx/issues/11622

**Comment by tiagodread** [comment]
Fixed with the merge of #11839
